src/rt64lib/shaders/Surface.hlsl

```diff
@@ -85,9 +87,21 @@ void SurfaceAnyHit(inout HitInfo payload, Attributes attrib) {
 				vertex.normal = -vertex.normal;
 			}
 
+			// Sample the specular map.
+			int specularTexIndex = instanceProps[instanceId].materialProperties.specularTexIndex;
+			if (specularTexIndex >= 0) {
+				float uvDetailScale = instanceProps[instanceId].materialProperties.uvDetailScale;
+				specularColor = half(SampleTexture(gTextures[specularTexIndex], vertex.uv * uvDetailScale, instanceProps[instanceId].materialProperties.filterMode, instanceProps[instanceId].materialProperties.hAddressMode, instanceProps[instanceId].materialProperties.vAddressMode).r);
+			}
+			else {
+				float uvDetailScale = instanceProps[instanceId].materialProperties.uvDetailScale;
+				specularColor = (1.0h);
```

Comment:
uvDetailScale seems to be unused on this else branch. Perhaps move the declaration of specularColor to be 1.0h by default above the first if and ignore this else?